### PR TITLE
Reaction translation and single kick reaction

### LINKED ISSUE #3223: [Improvement]: OOM-kick-reaction
### Which feature

As far as I remember, the reaction tab had the function to react to OOM kicks and it doesn't seem to exist anymore since one of the updates.

### Reason

Now if someone wants to kick me with OOM, they will no longer be kicked automatically and only if they use other methods, which in turn works normally.

### Additional context

_No response_

### LINKED ISSUE #3375: [Request]: OOM Kick Reaction
### Problem

There's no way to set reactions to an OOM kick.

### Solution

Add OOM Kick to the reactions tab

### Reason

**My personal reason and what I've seen:**
This reaction is so important because it warns us that the attacker might be about to send us a desync kick as its last resort (predictable human behavior), and what does this means? if we enable this reaction we can instantly trigger a smart kick/block join against them before they have time to do anything else (like sending all their crash/kick methods towards us) (this happened to me yesterday and i couldn't find the guy on the player list, and got me desync kicked..)

Also the reactions not only are useful for auto kicking them, the auto-add to database feature is a must honestly.

Anyways, it's needed whatever the reason is, we got network bail and other useless kicks in reactions, why not OOM?

### Additional context

300px-Modern_Problems_Require_Modern_Solutions

### PR DESCRIPTION
**Pull request by ghost**
Closes #3375
Closes #3223
I also combined all reactions to kicks into one and added translations of event names.

### CODE REVIEW

**Comment by lonelybud** [comment]
i think it is good to know the specific kick that happened. You can make OOM kick a unique reaction.

Edit: This pr provides simplicity though from user experience perspective.

**Comment by gir489returns** [comment]
We can't approve this until you make a translation PR for at least the English. (The Russian translations would also be nice)

**Review by gir489returns — APPROVED** [review]

**Comment by snowleopardd** [comment]
Hey! thanks for addressing my issue, i've been testing this build but when someone tries to kick me it doesn't show the type of kick anymore and i'm getting a buggy notification ingame, doesn't even show the person who's triggering it, i guess it's the placeholder for translation? still weird

image

image

**Comment by ghost** [comment]
> Hey! thanks for addressing my issue, i've been testing this build but when someone tries to kick me it doesn't show the type of kick anymore and i'm getting a buggy notification ingame i guess it's the placeholder for translations image

are you using this [translation](https://github.com/YimMenu/Translations/pull/186)?

**Comment by snowleopardd** [comment]
> > Hey! thanks for addressing my issue, i've been testing this build but when someone tries to kick me it doesn't show the type of kick anymore and i'm getting a buggy notification ingame i guess it's the placeholder for translations image
> 
> are you using this [translation](https://github.com/YimMenu/Translations/pull/186)?

Yeah that was it! thx, though, i don't think merging all kicks into one notification is a good thing, it was useful to know the types

**Comment by ghost** [comment]
> Yeah that was it! thx, though, i don't think merging all kicks into one notification is a good thing, it was useful to know the types

nah this is useless. you block all those kicks anyway

File: src/core/settings.hpp [hunk 1]
```diff
@@ -202,57 +202,55 @@ namespace big
...
+			reaction send_to_cutscene{"SEND_TO_CUTSCENE", "REACTION_SEND_TO_CUTSCENE_NOTIFY", "REACTION_SEND_TO_CUTSCENE_ANNOUNCE"};
+			reaction send_to_location{"SEND_TO_LOCATION", "REACTION_SEND_TO_LOCATION_NOTIFY", "REACTION_SEND_TO_LOCATION_ANNOUNCE"};
+			reaction send_to_interior{"TP_TO_INTERIOR", "REACTION_SEND_TO_INTERIOR_NOTIFY", "REACTION_SEND_TO_INTERIOR_ANNOUNCE"};
+			reaction sound_spam{"SOUND_SPAM", "REACTION_SOUND_SPAM_NOTIFY", "REACTION_SOUND_SPAM_ANNOUNCE"};
+			reaction spectate_notification{"SPECTATE_NOTIFICATION", "REACTION_SPECTATE_NOTIFICATION_NOTIFY", "REACTION_SPECTATE_NOTIFICATION_ANNOUNCE"};
+			reaction give_collectible{"GIVE_COLLECTIBLE", "REACTION_GIVE_COLLECTIBLE_NOTIFY", "REACTION_GIVE_COLLECTIBLE_ANNOUNCE"};
+			reaction transaction_error{"SHOW_TRANSACTION_ERROR", "REACTION_TRANSACTION_ERROR_NOTIFY", "REACTION_TRANSACTION_ERROR_ANNOUNCE"};
+			reaction tse_freeze{"TSE_FREEZE", "REACTION_TSE_FREEZE_NOTIFY", "REACTION_TSE_FREEZE_ANNOUNCE"};
+			reaction tse_sender_mismatch{"TSE_FREEZE_MISMATCH", "REACTION_TSE_SENDER_MISMATCH_NOTIFY", "REACTION_TSE_SENDER_MISMATCH_ANNOUNCE"};
```
Comment (gir489returns) [inline src/core/settings.hpp:230]: Copypaste error.

**Review by gir489returns — CHANGES_REQUESTED** [review]

File: src/core/settings.hpp (hunk 1 above)
Comment (ghost) [reply src/core/settings.hpp]: do you mean `"TSE_FREEZE_MISMATCH"`?

Comment (gir489returns) [reply src/core/settings.hpp:230]: It's supposed to be tse_sender_mismatch.

File: src/backend/reactions/interloper_reaction.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ namespace big
 		{
 			LOGF(WARNING,
 			    "Received {} from {} ({}), victim is {}",
-			    m_event_name,
+			    g_translation_service.get_translation(m_event_name),
```
Comment (Yimura) [inline src/backend/reactions/interloper_reaction.cpp]: We don't translate logs as they're meant for developers.

File: src/backend/reactions/reaction.cpp [hunk 3]
```diff
@@ -57,7 +57,7 @@ namespace big
 
 		if (log)
 		{
-			LOGF(WARNING, "Received {} from {} ({})", m_event_name, player->get_name(), player->get_rockstar_id());
+			LOGF(WARNING, "Received {} from {} ({})", g_translation_service.get_translation(m_event_name), player->get_name(), player->get_rockstar_id());
```
Comment (Yimura) [inline src/backend/reactions/reaction.cpp]: See other comment.

File: src/backend/reactions/reaction.cpp [hunk 4]
```diff
@@ -92,7 +92,7 @@ namespace big
...
```
Comment (Yimura) [inline src/backend/reactions/reaction.cpp]: See other comment.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — APPROVED** [review]

**Review by gir489returns — APPROVED** [review]

**Pull request merged**